internal/extension/oop.go

```diff
@@ -214,31 +212,53 @@ func (p *OOPExtension) monitorStderr(stderr io.ReadCloser, stopChan <-chan struc
 	}
 }
 
-func (p *OOPExtension) logStderr(logLvl logLevel, logString string, err error) {
-	switch logLvl {
+func (p *OOPExtension) logStderr(logLine *oopLogEntry) {
+	keysAndValues := make([]any, 0, len(logLine.KeysAndValues)*2)
+	for key, value := range logLine.KeysAndValues {
+		keysAndValues = append(keysAndValues, key, value)
+	}
+	switch logLine.Level {
 	case LogLevelInfo:
-		p.logger.Info(logString)
+		p.logger.Info(logLine.Msg, keysAndValues...)
+	case LogLevelError:
+		p.logger.Error(fmt.Errorf("%v", logLine.Error), logLine.Msg, keysAndValues...)
 	default:
-		p.logger.Error(err, logString)
+		p.logger.Error(fmt.Errorf("unknown LogLevel %v", logLine.Level), logLine.Msg, keysAndValues...)
 	}
 }
 
-func parseStderr(logLine []byte) (logLvl logLevel, logString string, err error) {
-	if len(logLine) == 0 {
-		return LogLevelError, "", fmt.Errorf("input byte slice is empty")
-	}
+type logLevel string
 
-	// Convert first byte to integer and validate log range
-	logLvl = logLevel(logLine[0])
-	if logLvl < LogLevelInfo || logLvl > LogLevelError {
-		// At the moment only differencing from Info and Error
-		return LogLevelError, "", fmt.Errorf("first byte is not a valid log level range 0-1. log: %q", string(logLine))
-	}
+const (
+	LogLevelInfo  logLevel = "info"
+	LogLevelError logLevel = "error"
+)
```

Comment:
Should we reuse [this](https://github.com/Kuadrant/kuadrant-operator/blob/main/internal/log/log.go#L41-L58) ?